### PR TITLE
Bugfix/live 10341 LLD polish of "locked device" device action state

### PR DESCRIPTION
**Pull request by ofreyssinet-ledger**
#### before
Screenshot 2023-12-19 at 17 05 45

#### after 
Screenshot 2023-12-19 at 17 04 36

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10341] 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - Fix "locked device" device action state rendering
  - Rework `ErrorBody` component used by device actions so that it works everywhere even in v2 style provider
  - Use `ErrorBody` for "device not onboarded" device action state

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10341]: https://ledgerhq.atlassian.net/browse/LIVE-10341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/** Renders an error icon, title and description */
+export const ErrorBody: React.FC<{
+  /**
+   * react node to render instead of the icon, at the top
+   * */
+  top?: React.ReactNode | null | undefined;
+  /**
+   * icon to render inside a box
+   */
+  Icon?:
```
Comment (aussedatlo) [inline apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:26]: [ASK]: Why `Icon` is the only props with a uppercase ?

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  whiteSpace: "pre-wrap",
+  textAlign: "center",
+})``;
+
+/** Renders an error icon, title and description */
+export const ErrorBody: React.FC<{
+  /**
+   * react node to render instead of the icon, at the top
+   * */
+  top?: React.ReactNode | null | undefined;
```
Comment (aussedatlo) [inline apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:22]: For enhanced readability, it is possible to place optional properties after required properties ?

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { BoxedIcon, Text } from "@ledgerhq/react-ui";
+import { withV3StyleProvider } from "~/renderer/styles/StyleProviderV3";
+import styled from "styled-components";
+
+export const ErrorTitle = styled(Text).attrs({
+  color: "neutral.c100",
+  textAlign: "center",
+})``;
```
Comment (aussedatlo) [inline apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:9]: [ASK]: what is the benefit to use `attrs` to edit style ?

you can also do something like

```ts
export const ErrorTitle = styled(Text)`
color: ${p => p.theme.colors.neutral.c100};
text-align: center;
`
```

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx (hunk 2 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:22]: yes that would make sense 👍 
we could try to push this as a convention to adopt

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx (hunk 1 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:26]: The type of `Icon` is a component type, so as a convention we uppercase the first letter

File: apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx (hunk 3 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/components/ErrorBody.tsx:9]: yes I think you're right, let me change this

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx [hunk 4]
```diff
@@ -433,7 +433,7 @@ export const DeviceActionDefaultRendering = <R, H extends States, P>({
     // NB Until we find a better way, remap the error if it's 6d06 (LNS, LNSP, LNX) or 6d07 (Stax) and we haven't fallen
     // into another handled case.
     if (isDeviceNotOnboardedError(e)) {
-      return <DeviceNotOnboardedError t={t} device={device} />;
+      return <DeviceNotOnboardedErrorComponent t={t} device={device} />;
```
Comment (mbertin-ledger) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx:436]: THX 👍

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 5]
```diff
@@ -1753,7 +1753,7 @@
   "DeviceAction": {
     "allowAppPermission": "Open the {{wording}} app on your device",
     "allowAppPermissionSubtitleToken": "to manage your {{token}} tokens",
-    "allowManagerPermission": "Allow Ledger Manager on your device",
+    "allowManagerPermission": "Allow Ledger Manager on your {{productName}}",
```
Comment (mbertin-ledger) [inline apps/ledger-live-desktop/static/i18n/en/app.json:1756]: [ASK] Will it change after next Smartling pull ? Moreover we need to know if the other translation contains the placeholders with the same attribute name ?

**Review by aussedatlo — APPROVED** [review]

**Pull request merged**